### ISSUE TITLE
Rule Request: Legacy Multiple

### ISSUE THREAD

**Issue by marcelofabri** [realm/SwiftLint#2612]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

Prefer using `x.isMultiple(of: y)` than `x % y == 0`.

1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.

https://github.com/apple/swift-evolution/blob/master/proposals/0225-binaryinteger-iseven-isodd-ismultiple.md

2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

**Would trigger**
```swift
// UITableView alternating row colour
cell.contentView.backgroundColor = indexPath.row % 2 == 0 ? .gray : .white

// Codable.swift.gyb in apple/swift
guard count % 2 == 0 else { throw DecodingError.dataCorrupted(...) }

// Bool.swift in apple/swift
public static func random<T: RandomNumberGenerator>(using generator: inout T) -> Bool {
    return (generator.next() >> 17) & 1 == 0
}

// KeyPath.swift in apple/swift
_sanityCheck(bytes > 0 && bytes % 4 == 0, "capacity must be multiple of 4 bytes")

// ReversedCollection Index.base documentation https://developer.apple.com/documentation/swift/reversedcollection/index/2965437-base
guard let i = reversedNumbers.firstIndex(where: { $0 % 2 == 0 })
```

**Wouldn't trigger**

```swift
// UITableView alternating row colour
cell.contentView.backgroundColor = indexPath.row.isMultiple(of: 2) ? .gray : .white

// Codable.swift.gyb in apple/swift
guard count.isMultiple(of: 2) else { throw DecodingError.dataCorrupted(...) }

// Bool.swift in apple/swift
public static func random<T: RandomNumberGenerator>(using generator: inout T) -> Bool {
    return (generator.next() >> 17).isMultiple(of: 2)
}

// KeyPath.swift in apple/swift
_sanityCheck(bytes > 0 && bytes.isMultiple(of: 4), "capacity must be multiple of 4 bytes")

// not used in a boolean context
let secretValue = (value % 3) + 2
```

3. Should the rule be configurable, if so what parameters should be configurable?

Just severity

4. Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a rule as opt-in.

Opt-in as `isMultiple(of:)` is only available on Swift 5+.
